src/lando/api/tests/test_git.py

```diff
@@ -0,0 +1,81 @@
+import subprocess
+
+import pytest
+
+from lando.api.tests.conftest import git_setup_user
+from lando.main.scm.git import GitSCM
+
+
+# We can't use fixtures directly in parametrize [0],
+# so we jumps through some hoops.
+# [0] https://github.com/pytest-dev/pytest/issues/349
+@pytest.mark.parametrize(
+    "path,repo_fixture_name,expected",
+    (
+        ("/non-existent-path", None, False),
+        ("/tmp", None, False),
+        (None, "git_repo", True),
+    ),
+)
+def test_is_initialised(path, repo_fixture_name, expected, request):
+    if not path:
+        path = request.getfixturevalue(repo_fixture_name).strpath
+    scm = GitSCM(path)
+    assert scm.repo_is_initialized == expected
```

Comment:
This test is hard to read. I wrote an alternate suggestion here...

```suggestion
@pytest.mark.parametrize(
    "path,repo_fixture_name,expected",
    (
        ("/non-existent-path", False),
        ("/tmp", False),
        (None, True),  # generate a default path 
    ),
)
def test_is_initialised(path, expected, request, git_repo):
    if not path:
        path = git_repo.strpath
    scm = GitSCM(path)
    assert scm.repo_is_initialized == expected
```

Though ideally, this test would be rewritten to more explicitly show a git repo (along with a path) being initialized before it is checked (instead of just defaulting to a `None` path.)

Reply:
Cool, I took that.

Ideally, we should be able to use `parametrize` with a fixture, but it's still an open isse for pytest https://github.com/pytest-dev/pytest/issues/349